test/tbb/test_concurrent_lru_cache.cpp

```diff
@@ -125,3 +125,27 @@ TEST_CASE("basic test for eviction of only unused items 2") {
     REQUIRE_MESSAGE(is_correct, "cache should not evict items in use");
 }
 
+//! \brief \ref error_guessing
+TEST_CASE("basic test for handling case when number_of_lru_history_items is zero") {
+    int call_count = 0;
+    auto counter_func = [&call_count] (int key) {
+        call_count++;
+        return key*2;
+    };
+    
+    using cache_type =  tbb::concurrent_lru_cache<int, int, decltype(counter_func)>;
+    cache_type cache{counter_func, 0};
+
+    for(int i = 1; i <= 10; ++i) {
+        cache[1];
+        REQUIRE_MESSAGE(call_count == i, "when number_of_lru_history_items is zero, element must be erased after use");
```

Comment:
As for me, this check does not guarantee that the element is erased from the cache. Since we don't have the API that checks for the element presence in the cache, I would recommend to use an object with custom destructor (or check other tests and use some pre-existing type with the static counter) to check if the object was actually deleted by calling the dtor.